src/index.ts

```diff
@@ -0,0 +1,125 @@
+
+const MONTHS_IN_YEAR = 12;
+const MINIMUM_PURCHASE_AMOUNT = 800;
+
+export type Result = {
+  year: number;
+  depreciationMonths: number;
+  depreciationAmount: number;
+  percentage: number;
+  startValue: number;
+  endValue: number;
+}
+
+const assertPurchaseAmount = (amount: number) => {
+  if (amount < 0) {
+    throw new TypeError('`purchaseAmount` is invalid.');
+  }
+};
+
+const assertPurchaseDate = (dateString: string) => {
+  if (isNaN(new Date(dateString).valueOf())) {
+    throw new TypeError('`purchaseDate` is invalid.');
+  }
+};
+
+const assertDepreciationYears = (years: number) => {
+  if (years < 1) {
+    throw new TypeError('`totalDepreciationYears` is invalid.');
+  }
+};
+
+const toFixedTwo = (num: number): number => Math.round((num + Number.EPSILON) * 100) / 100;
```

Comment:
❓ Why do we need to add `Number.EPSILON` to the `num`?

Reply:
Because Math.round sometimes is incorrect, for example:

```
Math.round(1.005 * 100) / 100 === 1
Math.round((1.005 + Number.EPSILON) * 100) / 100 === 1.01
```

Reply:
If this is really needed then I'd at least put a comment for it

Reply:
Thank, I added a comment and some tests for `toFixedTwo` here: https://github.com/kontist/asset-depreciation-calculator/pull/1/commits/16d6507f791cb0451fb72d4002bdc32b0879a1c8

Reply:
I first time see that approach to fix rounding in JS :)
It's funny that even `toFixed` does rounding "incorrectly" for `1.005` 🤷 
```
(1.005).toFixed(2) === "1.00"
```

Reply:
`1.009.toFixed(2) === "1.01"` 🤷

Reply:
technically speaking, `0.05` in javascript is actually `0.04999999...`, that's why.

Reply:
`Number.EPSILON` is to add the last `0.00...001`.